lms/djangoapps/courseware/views.py

```diff
@@ -862,6 +876,59 @@ def get_cosmetic_display_price(course, registration_price):
         return _('Free')
 
 
+@require_global_staff
+@ensure_valid_course_key
+def enroll_staff(request, course_id):
+    """
+    1. Should be staff
+    2. should be a valid course_id
+    3. shouldn't be enrolled before
+    4. The requested view url to redirect
+
+    GET
+        html: return enroll staff page
+
+    POST
+        1. You want to register for this course?
+            Confirm:
+                1. User is valid staff user who wants to enroll.
+                2. Course is valid course
+        2. Yes
+        3. Post request, enroll the user and redirect him to the requested view
+
+    :param request:
+    :param course_id:
+    :return:
+    """
+    user = request.user
+    course_key = CourseKey.from_string(course_id)
+    _next = urllib.quote_plus(request.GET.get('next', 'info'), safe='/:?=')
+
+    if request.method == 'GET':
+        with modulestore().bulk_operations(course_key):
```

Comment:
@awaisdar001 `CourseSummary` returns the list of course and here we need to check only one course.